src/Features/CSharp/Portable/CodeRefactorings/SyncNamespace/CSharpSyncNamespaceService.cs

```diff
@@ -0,0 +1,348 @@
+
+                    if (namespaceDeclaration.Name.Span.IntersectsWith(position)
+                        && namespaceDeclaration.Name.GetDiagnostics().All(diag => diag.DefaultSeverity != DiagnosticSeverity.Error))
+                    {
+                        return namespaceDeclaration;
+                    }
+                }
+                else if (namespaceDecls.Length == 0)
+                {
+                    var firstMemberDeclarationName = compilationUnit.Members.FirstOrDefault().GetNameToken();
+
+                    if (firstMemberDeclarationName != default 
+                        && firstMemberDeclarationName.Span.IntersectsWith(position))
+                    {
+                        return compilationUnit;
+                    }
+                }
+
+                return null;
+            }
+        }
+
+        /// <summary>
+        /// Try to change the namespace declaration based on the refacoring rules:
+        ///     - if neither declared and target namespace are "" (i.e. global namespace),
+        ///     then we try to change the name of the namespace.
+        ///     - if declared namespace is "", then we try to move all types declared 
+        ///     in global namespace in the document into a new namespace declaration.
+        ///     - if target namespace is "", then we try to move all members in declared 
+        ///     namespace to global namespace (i.e. remove the namespace declaration).    
+        /// </summary>
+        protected override CompilationUnitSyntax ChangeNamespaceDeclaration(
+            CompilationUnitSyntax compilationUnit, 
+            ImmutableArray<string> declaredNamespaceParts, 
+            ImmutableArray<string> targetNamespaceParts)
+        {
+            Debug.Assert(!declaredNamespaceParts.IsDefault && !targetNamespaceParts.IsDefault);
+
+            // Move everything from global namespace to a namespace declaration
+            if (declaredNamespaceParts.Length == 1 && declaredNamespaceParts[0].Length == 0)
+            {
+                var targetNamespaceDecl = SyntaxFactory.NamespaceDeclaration(
+                    name: CreateNameSyntax(targetNamespaceParts, aliasQualifier: null, targetNamespaceParts.Length - 1)
+                            .WithAdditionalAnnotations(WarningAnnotation),
+                    externs: default, 
+                    usings: default,
+                    members: compilationUnit.Members);
+                return compilationUnit.WithMembers(new SyntaxList<MemberDeclarationSyntax>(targetNamespaceDecl));
+            }
+
+            // We should have a single member which is a namespace declaration in this compilation unit.
+            var namespaceDeclaration = compilationUnit.DescendantNodes().OfType<NamespaceDeclarationSyntax>().Single();
+
+            // Move everything to global namespace
+            if (targetNamespaceParts.Length == 1 && targetNamespaceParts[0].Length == 0)
```

Comment:
I think I swa there is IsGlobalNamespace call somewhere? share the code?

Reply:
Will fix.